mcp/transport.go

```diff
@@ -375,6 +375,18 @@ func (t *ioConn) read(ctx context.Context, in *json.Decoder) (jsonrpc.Message, e
 	if err := in.Decode(&raw); err != nil {
 		return nil, err
 	}
+	// Read remaining data in the buffer.
```

Comment:
I'd think this would be simpler. Doesn't it suffice to check that the next byte is a newline?